gulpfile.js

```diff
@@ -1,10 +1,140 @@
+
+var bases = {
+    dev: {
+        root: './',
+        app: './app',
+        libs: './app/libs',
+        assets: './assets',
+        testapi: './testapi'
+    },
+    dist: {
+        root: 'dist/',
+        app: 'dist/app/',
+        libs: 'dist/libs/',
+        assets: 'dist/assets',
+        testapi: 'dist/testapi'
+    }
+};
+
+var src_paths = {
+    index: [
+        './index.html',
+        './app.js',
+        './main.css',
+        './config.js'
+    ],
+    libs: [
+        './bower_components/angular/angular.js',
+        './bower_components/leaflet/leaflet.*',
+        './bower_components/**/angular-animate.js',
+        './bower_components/**/angular-aria.js',
+        './bower_components/**/angular-leaflet-directive.js',
+        './bower_components/**/angular-material.js',
+        './bower_components/**/angular-material.css',
+        './bower_components/**/angular-ui-router.js',
+        './bower_components/**/ngStorage.js',
+    ],
+    assets: ['./assets/**/*'],
+    components: ['./app/**/*'],
+    testapi: ['./testapi/**/*']
+};
+
+var dist_paths = {
+    css: [
+    bases.dist.root + 'assets/**/*.css',
+    bases.dist.root + 'libs/**/*.css',
+    bases.dist.root + 'main.css'
+    ],
+    js: [
+        bases.dist.root + 'assets/**/*.js',
+        bases.dist.root + 'libs/angular.*js',
+        bases.dist.root + 'libs/**/*.js',
+        bases.dist.root + 'app.js',
+        bases.dist.root + 'app/components/**/*.js'
+    ]
+};
+
+function dist_copy () {
+    gulp.src(src_paths.index)
+        .pipe(gulp.dest(bases.dist.root));
+    gulp.src(src_paths.libs)
+        .pipe(flatten())
+        .pipe(gulp.dest(bases.dist.libs));
+    gulp.src(src_paths.assets)
+        .pipe(gulp.dest(bases.dist.assets));
+    gulp.src(src_paths.components)
+        .pipe(gulp.dest(bases.dist.app));
+    gulp.src(src_paths.testapi)
+        .pipe(gulp.dest(bases.dist.testapi));
+
+}
+
+function inject_imports ($envType) {
+    if ($envType === 'dist') {
+        var dist_index = gulp.src(bases.dist.root + 'index.html');
+
+        var dist_includes = gulp.src(dist_paths.css.concat(dist_paths.js), {read:false});
+
+        // Flush the inject blocks so the real include can work
+        dist_index.pipe(inject(gulp.src('')))
+            .pipe(gulp.dest(bases.dist.root));
+
+        // Include really now
+        dist_index.pipe(inject(dist_includes, {addRootSlash: false, ignorePath: bases.dist.root}))
+            .pipe(gulp.dest(bases.dist.root));
+    } else {
+        var dev_index = gulp.src('./index.html');
+        // It's not necessary to read the files (will speed up things), we're only after their paths:
+        var dev_includes = gulp.src(
+            src_paths.assets.concat(
+                src_paths.libs).concat(
+                src_paths.index).concat(
+                src_paths.components), {read: false});
+
+        dev_index.pipe(inject(dev_includes, {addRootSlash: false, ignorePath: '/'}))
```

Comment:
Line is too long.<br>Identifier 'dev_index' is not in camel case.<br>Identifier 'dev_includes' is not in camel case.